x/proof/keeper/msg_server_create_claim.go

```diff
@@ -77,17 +78,36 @@ func (k msgServer) CreateClaim(
 	if err != nil {
 		return nil, status.Error(codes.Internal, types.ErrProofInvalidClaimRootHash.Wrap(err.Error()).Error())
 	}
-	numComputeUnits, err = claim.GetNumComputeUnits()
+
+	numClaimComputeUnits, err = claim.GetNumComputeUnits()
 	if err != nil {
 		return nil, status.Error(codes.Internal, types.ErrProofInvalidClaimRootHash.Wrap(err.Error()).Error())
 	}
-	_, isExistingClaim = k.Keeper.GetClaim(ctx, claim.GetSessionHeader().GetSessionId(), claim.GetSupplierOperatorAddress())
 
-	// TODO_UPNEXT(#705): Check (and test) that numClaimComputeUnits is equal
-	// to num_relays * the_compute_units_per_relay for this_service.
-	// Add a comment that for now, we expect it to be the case because every
-	// relay for a specific service is wroth the same, but may change in the
-	// future.
+	// For now we expect the following equation to always hold:
+	// numClaimComputeUnits = numRelays * service.computeUnitsPerRelay
+	// This is because for any specific service, every relay is worth the same.
+	// However, this may change in the future.
+	serviceCupr, err := k.getServiceCupr(ctx, claim.SessionHeader.Service.Id)
+	if err != nil {
+		return nil, status.Error(codes.Internal, types.ErrProofServiceNotFound.Wrap(err.Error()).Error())
```

Comment:
I think this should be:
```suggestion
		return nil, status.Error(codes.NotFound, types.ErrProofServiceNotFound.Wrapf("%v", err).Error())
```